flexmeasures/api/common/utils/deprecation_utils.py

```diff
@@ -14,32 +14,32 @@ def sunset_blueprint(
     api_version_sunset: str,
     sunset_link: str,
     api_version_upgrade_to: str = "3.0",
-    blueprint_contents_removed: bool = True,
+    rollback_possible: bool = True,
 ):
     """Sunsets every route on a blueprint by returning 410 (Gone) responses, if sunset is active.
 
     Whether the sunset is active can be toggled using the config setting "FLEXMEASURES_API_SUNSET_ACTIVE".
-    If inactive, either:
-    - return 404 (Not Found) if the blueprint contents have been removed, or
-    - pass the request to be handled by the endpoint implementation.
+    If inactive, pass the request to be handled by the endpoint implementation.
+    If the endpoint implementations have been removed, set rollback_possible=False.
 
     Errors will be logged by utils.error_utils.error_handling_router.
     """
 
     def let_host_switch_to_returning_410():
```

Comment:
The main purpose of this logic (as per the docstring) is to return 410 on all routes.

The possible exception is in the else clause.

1. Let's rename this function, e.g. to `return_410_sunset_unless_host_rises_the_sun`
2. I'd start with an if statement which inspects if we can let the requests do their work (`pass`). The 410 logic happens after this if-check, if we did not `pass`. I hope the resulting if condition would be easier to grasp (if sunset active and `rollback_possible`).